CvGameCoreDLL_Expansion2/CvPlot.cpp

```diff
@@ -15404,6 +15404,160 @@ int CvPlot::GetNumSpecificPlayerUnitsAdjacent(PlayerTypes ePlayer, const CvUnit*
 ///-------------------------------------
 
 #if defined(MOD_BALANCE_CORE)
+int CvPlot::GetStrategicValue(PlayerTypes ePlayer) const
+{
+	TeamTypes eTeam = GET_PLAYER(ePlayer).getTeam();
+	if (eTeam == NO_TEAM)
+		return 0;
+
+	CvDiplomacyAI* pDiplomacyAI = GET_PLAYER(ePlayer).GetDiplomacyAI();
+
+	// Evaluate based on surrounding plots
+	int iAdjacentUnownedLand = 0;
+	int iNearbyUnownedLand = 0;
+
+	int iAdjacentThreat = 0;
+	int iNearbyThreat = 0;
+
+	int iAdjacentTarget = 0;
+	int iNearbyTarget = 0;
+
+	int iAdjacentCoast = 0;
+	int iNearbyCoast = 0;
+
+	// Directly adjacent tiles
+	for (int iI = 0; iI < NUM_DIRECTION_TYPES; iI++)
+	{
+		CvPlot* pLoopAdjacentPlot = plotDirection(getX(), getY(), ((DirectionTypes)iI));
+
```

Comment:
I would suggest
`if (pLoopAdjacentPlot == NULL) continue;`
to reduce nesting.

IMHO
Also what about some functions because code here duplicates like 3 times... Something like

`int getApproachMultiplier(CivApproachTypes eApproach)`
`int getStrengthMultipler(StrengthTypes eStrengthRelativeToUs)`

Furthermore, looks like `if (!pLoopNearbyPlot->isOwned())` whole block is also repeated two times so maybe put into other function..

Reply:
Yes, that function got a bit messy :)